pkg/audit/reporter.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+
+	"storj.io/storj/pkg/auth"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/provider"
+	dbx "storj.io/storj/pkg/statdb/dbx"
+	proto "storj.io/storj/pkg/statdb/proto"
+	"storj.io/storj/pkg/statdb/sdbclient"
+)
+
+type reporter interface {
+	RecordFailedAudits(ctx context.Context, failedNodes []*pb.Node) (err error)
+}
+
+// Reporter records audit reports in statdb and implements the reporter interface
+type Reporter struct {
+	statdb sdbclient.Client
+}
+
+// NewReporter instantiates a reporter
+func NewReporter(ctx context.Context, statDBPort string) (reporter *Reporter, err error) {
+	ca, err := provider.NewCA(ctx, 12, 14)
+	if err != nil {
+		return nil, err
+	}
+	identity, err := ca.NewIdentity()
+	if err != nil {
+		return nil, err
+	}
+	apiKey, ok := auth.GetAPIKey(ctx)
+	if !ok {
+		return nil, Error.New("invalid API credentials")
+	}
+
+	client, err := sdbclient.NewClient(identity, statDBPort, apiKey)
+	if err != nil {
+		return nil, err
+	}
+	return &Reporter{statdb: client}, nil
+}
+
+// RecordFailedAudits saves failed audit details to statdb
+func (reporter *Reporter) RecordFailedAudits(ctx context.Context, failedNodes []*pb.Node) (err error) {
```

Comment:
I think `RecordFailedAudits` should be called `RecordAudits`, and the nodes passed in should already have things like `UpdateUptime`, `isUp`, etc... set based on audit results. That way you can keep most of your code here the same and use this function for everything, and not just failed audits.